main.js

```diff
@@ -171,3 +171,71 @@ if (closeIcon) {
     navMenu.classList.remove('active');
   });
 }
+
+// form validations functionalty codes are here
+
+const nameError = document.getElementById('name-error');
+const emailError = document.getElementById('email-error');
+const messageError = document.getElementById('message-error');
+const sumbitError = document.getElementById('submit-error');
+
+function nameValidation() {
+  const name = document.getElementById('contact-name').value;
+  if (name.length === 0) {
+    nameError.innerHTML = 'insert your name';
+    return false;
+  }
+  if (!name.match(/^[a-zA-Z]+ [a-zA-Z]+$/)) {
+    nameError.innerHTML = 'insert full name';
+    return false;
+  }
+  nameError.innerHTML = '<i class="bi bi-check-circle-fill"></i>';
+  return true;
+}
+
+function emailValidation() {
+  const email = document.getElementById('contact-email').value;
+  if (email.length === 0) {
+    emailError.innerHTML = 'insert your email';
+    return false;
+  }
+  if (!email.match(/^(([^<>()\]\\.,;:\s@"]+(\.[^<>()\]\\.,;:\s@"]+)*)|(".+"))@(([0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/)) {
+    emailError.innerHTML = 'Invalid Email';
+    return false;
+  }
+  emailError.innerHTML = '<i class="bi bi-check-circle-fill"></i>';
+  return true;
+}
+function messageValidation() {
+  const message = document.getElementById('contact-message').value;
+  const required = 30;
+  const left = required - message.length;
+  if (left > 0) {
+    messageError.innerHTML = `${left} More characters required`;
+    return false;
+  }
+  messageError.innerHTML = '<i class="bi bi-check-circle-fill"></i>';
+  return true;
+}
+function formValidation() {
+  const emailField = document.getElementById('contact-email').value;
+  const emailRegex = /^[a-z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/;
+  if (!nameValidation() || !emailValidation() || !messageValidation()) {
+    sumbitError.style.display = 'block';
+    sumbitError.innerHTML = 'Please fill all fields correctly';
+    setTimeout(() => {
+      sumbitError.style.display = 'none';
+    }, 3000);
+    return false;
+  }
+  if (!emailField.match(emailRegex)) {
+    sumbitError.style.display = 'block';
+    sumbitError.innerHTML = 'Email field should be in lower case';
+    setTimeout(() => {
+      sumbitError.style.display = 'none';
+    }, 3000);
+    return false;
+  }
+  return true;
+}
+formValidation();
```

Comment:
- [ ]  Kindly as requested by the previous reviewer note that the required validation is this.

- The contact form includes a validation that checks if the email is written in all lowercase.
- If the user submits the contact form with any uppercase characters in the email field, an error message will appear.
- However, if the email field only contains lowercase characters, the form will be sent normally.
- The error message will be displayed near the submit button or the input fields.
- To check if the email is all lowercase, you can compare the email and its lowercase version, such as email === email.toLowerCase().
- You can also use regular expressions or loop through each letter of the email string.
- Any implementation of this check is considered valid as long as it works correctly.
- The function that checks the email format is attached to the form "submit" or the submit button/input "click" event. kindly consider following all the requirements above one by one to deliver the expected output